nservicebus/messaging/moving-handlers.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Moving Handlers
+summary: Guidelines on how to move handlers between the endpoints
+reviewed: 2021-09-01
+component: Core
+isLearningPath: true
+---
+
+As distributed systems evolve, the need to change which endpoint handles a message may rise. During the development phase, such changes are relatively straightforward, but when the system is in production additional considerations need to be given to messages that are in-flight. 
+
+## How to move Handlers in between Endpoints
+
+To move a handler from `SourceEndpoint` to a `DestinationEndpoint`, the following steps need to be taken:
+
+ - Copy the handler into `DestinationEndpoint` and deploy that endpoint. Now both endpoints are handling the same message type.
+ - Make sure that `DestinationEndpoint` receives the messages that it should be handling. From now on all newly sent messages should only go to the DestinationEndpoint.
+ - After ensuring that the `SourceEndpoint` does not have any remaining messages of the type that should be processed by `DestinationEndpoint`, delete the handler from `SourceEndpoint` and deploy it.
+
+### How to ensure that the messages get to the DestinationEndpoint
+
+This depends on whether the message is a command or an event. 
+
+When the handler process events, it is important for the `DestinationEndpoint` to subscribe to that event. That will ensure that this event is delivered to `DestinationEndpoint`. Followed by that, `SourceEndpoint` should unsubscribe from that event to ensure that no new events are delivered to the `SourceEndpoint` queue.
+
+When the handler processes commands, every piece of code that sends that command needs to be updated to send them to a queue of `DestinationEndpoint`. In the meantime, the `SourceEndpoint`-handler may be changed to [forward the messages to the new destination](/nservicebus/messaging/forwarding.md). 
+
+### How to handle error messages that were discovered after removing the handler
```

Comment:
Excellent that you did not miss this! Well done! :-)

Reply:
It is Murphy's law: If something can go wrong, it will go wrong